apps/vs-code-designer/src/app/commands/createCodeful/CodefulWorkflowCreateStep.ts

```diff
@@ -0,0 +1,148 @@
+  private constructor() {
+    super();
+  }
+
+  public static async createStep(context: IActionContext): Promise<CodefulWorkflowCreateStep> {
+    await validateDotnetInstalled(context);
+    return new CodefulWorkflowCreateStep();
+  }
+
+  public async executeCore(context: IFunctionWizardContext): Promise<string> {
+    const functionPath: string = path.join(context.projectPath, nonNullProp(context, 'functionName'));
+
+    const codelessDefinition: string = await getCodefulWorkflowTemplate();
+
+    const workflowCsFullPath: string = path.join(functionPath, codefulWorkflowFileName);
+
+    await fse.ensureDir(functionPath);
+    await fse.writeFile(workflowCsFullPath, codelessDefinition);
+
+    await this.createSystemArtifacts(context);
+
+    await createConnectionsJson(context, context.projectPath);
+
+    return workflowCsFullPath;
+  }
+
+  private async createSystemArtifacts(context: IFunctionWizardContext): Promise<void> {
+    const hostJsonPath: string = path.join(context.projectPath, hostFileName);
+    let hostJson: IHostJsonV2 = await this.getHostJson(context, hostJsonPath);
+    let hostJsonUpdated = false;
+
+    const target = vscode.Uri.file(context.projectPath);
+
+    await switchToDotnetProject(context, target, '8', true);
+
+    hostJson.extensionBundle = undefined;
+    hostJsonUpdated = true;
+
+    if (
+      nonNullProp(context, 'workflowProjectType') === WorkflowProjectType.Bundle &&
+      (hostJson.extensionBundle === undefined ||
+        hostJson.extensionBundle.id !== extensionBundleId ||
+        hostJson.extensionBundle.version !== defaultVersionRange)
+    ) {
+      hostJson = {
+        ...hostJson,
+      };
+      hostJson.extensionBundle = undefined;
+      hostJsonUpdated = true;
+    }
+
+    if (hostJsonUpdated) {
+      await writeFormattedJson(hostJsonPath, hostJson);
+    }
+    await setLocalAppSetting(context, context.projectPath, workerRuntimeKey, WorkerRuntime.Dotnet, MismatchBehavior.Overwrite);
+    await setLocalAppSetting(context, context.projectPath, functionsInprocNet8Enabled, '1', MismatchBehavior.Overwrite);
+    await setLocalAppSetting(context, context.projectPath, appKindSetting, logicAppKind, MismatchBehavior.Overwrite);
+    await setLocalAppSetting(
+      context,
+      context.projectPath,
+      azureWebJobsStorageKey,
+      localEmulatorConnectionString,
+      MismatchBehavior.Overwrite
+    );
+  }
+
+  private async getHostJson(context: IFunctionWizardContext, hostJsonPath: string, allowOverwrite = false): Promise<IHostJsonV2> {
+    return this.getJsonFromFile(context, hostJsonPath, { version: '2.0' }, allowOverwrite);
+  }
+
+  private async getJsonFromFile<T extends object>(
+    context: IFunctionWizardContext,
+    filePath: string,
+    defaultValue: T,
+    allowOverwrite = false
+  ): Promise<T> {
+    if (await fse.pathExists(filePath)) {
+      const data: string = (await fse.readFile(filePath)).toString();
+      if (emptyStringTest.test(data)) {
```

Comment:
Is the regex needed for this? maybe just .length()?

Reply:
I actually didn't write this but copied it over. I did consolidate it but I don't feel comfortable changing it now.